cluster/src/main/scala/org/apache/spark/sql/SnappySessionFactory.scala

```diff
@@ -98,14 +76,37 @@ trait SnappySQLJob extends SparkJobBase {
     }
   }
 
-  private def cleanJobConfig(c: Config): Config = {
+  def cleanJobConfig(c: Config): Config = {
     // TODO Remove snappydata properties path when available
     var sJobConfig = c.withoutPath(Constant.STORE_PROPERTY_PREFIX + com.pivotal.gemfirexd
       .Attribute.USERNAME_ATTR)
     sJobConfig = sJobConfig.withoutPath(Constant.STORE_PROPERTY_PREFIX + com.pivotal
       .gemfirexd.Attribute.PASSWORD_ATTR)
     sJobConfig
   }
+}
+
+
+trait SnappySQLJob extends SparkJobBase {
+  type C = Any
+
+  final override def validate(sc: C, config: Config): SparkJobValidation = {
+    val parentLoader = org.apache.spark.util.Utils.getContextOrSparkClassLoader
+    val currentLoader = SnappyUtils.getSnappyStoreContextLoader(parentLoader)
+    Thread.currentThread().setContextClassLoader(currentLoader)
+    SnappyJobValidate.validate(isValidJob(sc.asInstanceOf[SnappySession], SnappySessionFactory
+        .cleanJobConfig(config)))
+  }
```

Comment:
Here I have moved the updateCredentials and cleanJobConfig from class to object so that it could be used outside.
I think the diff is so because git's diff algorithm first sees the trait being moved down rather than defs being moved up.